### PR TITLE
[GPU] New debug config: dry_run

### PR DESCRIPTION
**Pull request by isanghao**
### Details:
 - New debug config for GPU; dry-run
 - It is mainly for running dlbenchmark dry-run

### CODE REVIEW

File: inference-engine/thirdparty/clDNN/runtime/debug_configuration.cpp [hunk 1]
```diff
@@ -114,6 +115,7 @@ debug_configuration::debug_configuration()
     get_gpu_debug_env_var("DumpLayers", dump_layers);
     get_gpu_debug_env_var("DumpLayersDstOnly", dump_layers_dst_only);
     get_gpu_debug_env_var("DisableOnednn", disable_onednn);
+    get_gpu_debug_env_var("DryRun", dry_run);
```
Comment (yeonbok) [inline inference-engine/thirdparty/clDNN/runtime/debug_configuration.cpp:118]: how about more expressively,  "DryRunPath" or "DryRunGraphPath"?

File: inference-engine/thirdparty/clDNN/src/program.cpp [hunk 2]
```diff
@@ -422,10 +422,18 @@ void program::build_program(bool is_internal) {
...
-    compile();
-    init_kernels();
+
+    GPU_DEBUG_GET_INSTANCE(debug_config);
+#ifdef GPU_DEBUG_CONFIG
+    if (debug_config->dry_run.empty()) {
+#else
+    {
+#endif
+        { post_optimize_graph(is_internal); }
```
Comment (sshlyapn) [inline inference-engine/thirdparty/clDNN/src/program.cpp:432]: If `DryRun` option is enabled, then we skip some graph optimizations in `post_optimize_graph` stage. Is such behavior expected?

File: inference-engine/thirdparty/clDNN/runtime/debug_configuration.cpp (hunk 1 above)
Comment (isanghao) [reply inference-engine/thirdparty/clDNN/runtime/debug_configuration.cpp]: applied, thanks!

File: inference-engine/thirdparty/clDNN/src/program.cpp (hunk 2 above)
Comment (isanghao) [reply inference-engine/thirdparty/clDNN/src/program.cpp]: Oh, that was due to my ignorance :( Thanks for pointing out!

**Review by sshlyapn — APPROVED** [review]

**Pull request merged**